client/verta/verta/pipeline/_pipeline_step.py

```diff
@@ -0,0 +1,213 @@
+# -*- coding: utf-8 -*-
+
+from typing import Any, Dict, List, Optional
+
+from verta._internal_utils._utils import Configuration, Connection
+from verta.registry.entities import RegisteredModel, RegisteredModelVersion
+
+
+class PipelineStep:
+    """
+    A single step within an inference pipeline, representing a single model version to be run.
+
+    Parameters
+    ----------
+    name : str
+        Name of the step, for use within the scope of the pipeline only.
+    model_version : :class:`~verta.registry.entities.RegisteredModelVersion`
+        Registered model version to run for the step.
+    predecessors : list, optional
+        List of PipelineSteps whose outputs will be treated as inputs to this step.
+        If not included, the step is assumed to be an initial step. Values must be unique.
+
+    Attributes
+    ----------
+        name : str
+            Name of the step within the scope of the pipeline.
+        model_version
+            :class:`~verta.registry.entities.RegisteredModelVersion` run by this step.
+        predecessors : list
+            List of PipelineSteps whose outputs will be treated as inputs to this step.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        model_version: RegisteredModelVersion,
+        predecessors: Optional[
+            List["PipelineStep"]
+        ] = None,  # Optional because it could be the first step with no predecessors
+    ):
+        self._name = name
+        self._model_version = model_version
+        self._predecessors = predecessors or list()
```

Comment:
Let's re-use our own validation logic

```suggestion
        self._name = self.set_name(name)
        self._model_version = self.set_model_version(model_version)
        self._predecessors = self.set_predecessors(predecessors) if predecessors else list()
```